drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua

```diff
@@ -398,14 +396,19 @@ local function measurementHandlerFactory(capability_name, attribute, target_unit
       device:set_field(capability_name.."_unit", reporting_unit, {persist = true})
     end
 
+    local value, err_msg
     if reporting_unit then
-      local value = unit_conversion(ib.data.value, reporting_unit, target_unit)
-      device:emit_event_for_endpoint(ib.endpoint_id, attribute({value = value, unit = unit_strings[target_unit]}))
+      value, err_msg = unit_conversion(ib.data.value, reporting_unit, target_unit)
+    end
 
+    if value then
```

Comment:
If `value` is not `nil` but is equal to 0, wouldn't this be false? Would something like `if value ~= nil then` work instead?

Reply:
0 does not map to false in lua. Only `false` and `nil` map to false.